packages/import-bundle/test/test-compartment-wrapper.js

```diff
@@ -0,0 +1,150 @@
+/* global Compartment */
+import '@agoric/install-ses';
+import tap from 'tap';
+import { wrapInescapableCompartment } from '../src/compartment-wrapper.js';
+
+const { test } = tap;
+
+// We build a transform that allows oldSrc to increment the odometer, but not
+// read it. Note, of course, that SES provides a far easier way to accomplish
+// this (pass in a hardened `addMilage` endowment), but there are metering
+// use cases that don't involve voluntary participation by oldSrc.
+
+function milageTransform(oldSrc) {
+  if (oldSrc.indexOf('getOdometer') !== -1) {
+    throw Error(`forbidden access to 'getOdometer' in oldSrc`);
+  }
+  return oldSrc.replace(/addMilage\(\)/g, 'getOdometer().add(1)');
+}
+
+// add a lexical that the original code cannot reach
+function makeOdometer() {
+  let counter = 0;
+  function add(count) {
+    counter += count;
+  }
+  function reset() {
+    counter = 0; // forbidden
+  }
+  function read() {
+    return counter;
+  }
+  return { add, read, reset };
+}
+
+const createChild = `() => new Compartment({console})`;
+const doAdd = `addMilage()`;
+const doAddInChild = `(doAdd) => (new Compartment({console})).evaluate(doAdd)`;
+const attemptReset = `() => getOdometer().reset()`;
+const attemptResetInChild = `(attemptReset) => {
+  const c = new Compartment();
+  c.evaluate(attemptReset)();
+}`;
+
+// attempt to shadow 'getOdometer' (blocked by the inescapable lexical coming
+// last)
+const attemptResetByShadow = `(doAdd) => {
+  const taboo = 'get' + 'Odometer';
```

Comment:
Would javascript and our eslint settings allow you to use square single quotes on the outside and backticks on the inside here? That way reviewers wouldn't have to scan a multi-line string carefully in order to determine that there are no variable expansions inside.

Reply:
I don't think so.. backtick literals can include newlines, but not single-quote literals.

I could also build these as `function` definitions and then use them like:

```
`(${attemptResetByShadow})`
```

but that seemed harder to read.